app/controllers/api/queries_controller.rb

```diff
@@ -48,11 +50,20 @@ def show
     def create
       options = {}
       options[:filters] = build_filter
-
+      
       authorize_providers
+      start_date = params[:effective_start_date]
+      end_date = params[:effective_end_date]
+
+      rp = ReportingPeriod.where(start_date: start_date, end_date: end_date).first_or_create
+      rp.save!
 
-      options[:effective_date] = params[:effective_date]
+      options[:effective_start_date] = start_date
```

Comment:
assigning options[:effective_end_date] and options[:effective_start_date] is not necessary for the measure calculation. It just needs options['effective_date'] which should be the 'end_date'